### PR TITLE
wrap legacy code around core functionality

### PR DESCRIPTION
**Pull request by zerico007**

### CODE REVIEW

File: packages/legacy/src/react-to-webcomponent.test.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { describe, it, expect } from "vitest"
+import matchers from "@testing-library/jest-dom/matchers"
+
+// import r2wc from "./react-to-web-component"
+
+expect.extend(matchers)
+
+describe("core", () => {
```
Comment (christopherjbaker) [inline packages/legacy/src/react-to-webcomponent.test.tsx:8]: `decribe("legacy" ...)` (also suggests I left 'core' in place in react tests too

Comment (zerico007) [reply packages/legacy/src/react-to-webcomponent.test.tsx:8]: I'll update

File: packages/legacy/src/react-to-webcomponent.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  ReactElement,
+  RefObject,
+  Context,
+} from "@r2wc/core"
+import r2wcCore from "@r2wc/core"
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+const rootSymbol = Symbol.for("r2wc.root")
+
+interface ReactDOMType {
```
Comment (christopherjbaker) [inline packages/legacy/src/react-to-webcomponent.ts:15]: are these the same types from the legacy code?

Comment (zerico007) [reply packages/legacy/src/react-to-webcomponent.ts:15]: yes they are

**Pull request merged**